Create base scripts

surajssd opened the pull request.

surajssd committed:
coco-key-provider: Add configuration

- Add Dockerfile to build coco-key-provider image.
- Add script to build and push coco-key-provider image.

Signed-off-by: Suraj Deshmukh <suraj.deshmukh@microsoft.com>

surajssd committed:
kbs: Add script to deploy KBS

Signed-off-by: Suraj Deshmukh <suraj.deshmukh@microsoft.com>

surajssd committed:
encrypt-image: Add scripts to enable image encryption

- Add install-skopeo script.
- Add encrypt-image script.

Signed-off-by: Suraj Deshmukh <suraj.deshmukh@microsoft.com>

surajssd committed:
encrypted-app: Add deployment configuration

Signed-off-by: Suraj Deshmukh <suraj.deshmukh@microsoft.com>

surajssd committed:
Add instructions and Makefile

This commit adds instructions to encrypt image and deploy it on an
existing CAA setup.

Signed-off-by: Suraj Deshmukh <suraj.deshmukh@microsoft.com>

surajssd committed:
Add steps to install AKS & CAA

Signed-off-by: Suraj Deshmukh <suraj.deshmukh@microsoft.com>
Co-authored-by: Pradipta Banerjee <bpradipt@users.noreply.github.com>

bpradipt commented on README.md:
```diff
@@ -0,0 +1,56 @@
+# Deploying Encrypted Container Images with Cloud API Adatpor
```
```suggestion
# Deploying Encrypted Container Images with Cloud API Adaptor
```

bpradipt commented on README.md:
````diff
@@ -0,0 +1,56 @@
+
+```bash
+export KEY_FILE="$(pwd)/key.bin"
+head -c 32 /dev/urandom | openssl enc >"$KEY_FILE"
````
You can add a note here stating that the key needs to be 32 bytes. Afaik, this restriction is still there with KBS

bpradipt commented on README.md:
```diff
@@ -0,0 +1,56 @@
+
+### Encrypt Image
+
+Here we will use an existing `busybox` image encrypt it with the previously created key and upload it to a new destination.
```
```suggestion
Here we will use an existing `busybox` image, encrypt it with the previously created key and upload it to a new destination.
```

bpradipt commented on README.md:
````diff
@@ -0,0 +1,56 @@
+
+```bash
+export SOURCE_IMAGE=busybox
+export DESTINATION_IMAGE=quay.io/surajd/busybox-encrypted:$(date '+%Y-%m-%b-%d-%H-%M-%S')
````
```suggestion
export QUAY_USERID=REPLACE_ME
export DESTINATION_IMAGE=quay.io/${QUAY_USERID}/busybox-encrypted:$(date '+%Y-%m-%b-%d-%H-%M-%S')
```

bpradipt commented on README.md:
````diff
@@ -0,0 +1,56 @@
+
+```bash
+export AZURE_RESOURCE_GROUP=
+export SSH_KEY=
````
```suggestion
export SSH_KEY=REPLACE_ME
```

bpradipt commented on README.md:
````diff
@@ -0,0 +1,56 @@
+### Deploy AKS
+
+```bash
+export AZURE_RESOURCE_GROUP=
````
```suggestion
export AZURE_RESOURCE_GROUP=REPLACE_ME
```

bpradipt commented on README.md:
````diff
@@ -0,0 +1,56 @@
+Based on the output of the above command run:
+
+```bash
+export CLUSTER_SPECIFIC_DNS_ZONE=
````
```suggestion
export CLUSTER_SPECIFIC_DNS_ZONE=REPLACE_ME
```

bpradipt commented on deploy-infra/deploy-caa.sh:
```diff
@@ -0,0 +1,150 @@
+AZURE_SUBSCRIPTION_ID=$(az account show --query id --output tsv)
+AKS_RG="${AZURE_RESOURCE_GROUP}-aks"
+registry="quay.io/confidential-containers"
+AZURE_IMAGE_ID="/CommunityGalleries/cocopodvm-d0e4f35f-5530-4b9c-8596-112487cdea85/Images/podvm_image0/Versions/$(date -v -1d "+%Y.%m.%d" 2>/dev/null || date -d "yesterday" "+%Y.%m.%d")"
```
Do you want to pin an image version explicitly for reproducibility of the demo ?

bpradipt commented on deploy-infra/deploy-caa.sh:
```diff
@@ -0,0 +1,150 @@
+
+# Pull the CAA code
+if [ ! -d cloud-api-adaptor ]; then
+  git clone https://github.com/confidential-containers/cloud-api-adaptor
```
I think here also we can pin a specific commit of CAA. Since 0.8.0 is close, should we wait for the release and then pin to the 0.8.0 release ?

surajssd replied:
Right now pinning the latest main: b2b64c3c2de12dc25aa33e503d2036d2aca33204. I will test this today again to see if there are any issues.

bpradipt commented on encrypt-image/install-skopeo.sh:
```diff
@@ -0,0 +1,27 @@
+
+# Check if skopeo is already installed then skip this script
+if command -v skopeo >/dev/null 2>&1; then
+    echo "skopeo is already installed, skipping..."
```
We should verify the installed skopeo version. I have encountered issues where the installed one in the default path was older and didn't support encryption

surajssd replied:
done.

bpradipt commented on kbs/deploy-kbs.sh:
```diff
@@ -0,0 +1,77 @@
+
+# Pull the KBS code base if it is not available
+if [ ! -d kbs ]; then
+    git clone https://github.com/surajssd/kbs -b k8s-deploy-ingress-install
```
FIXME: switch to upstream kbs code

surajssd replied:
done

bpradipt commented on kbs/deploy-kbs.sh:
```diff
@@ -0,0 +1,77 @@
+
+# Set the image to use the latest bits from the KBS repository.
+pushd base
+kustomize edit set image kbs-container-image=quay.io/surajd/kbs:latest
```
FIXME: Switch to upstream released image. May be as part of pinning to 0.8.0 release?

surajssd replied:
Added.

bpradipt commented:
@surajssd great work on automating the demo scenario. I added few generic comments to help with adding this demo in the coco project and also making it reproducible by pinning the code and image versions used.

surajssd committed:
patch: Fix pulling encrypted container images

This commit uses components that have worked with encrypted images. The
issue is tracked in the upstream:
https://github.com/confidential-containers/cloud-api-adaptor/issues/1519.

Signed-off-by: Suraj Deshmukh <suraj.deshmukh@microsoft.com>

surajssd committed:
Add TODOs and pin versions

Signed-off-by: Suraj Deshmukh <suraj.deshmukh@microsoft.com>

surajssd committed:
install-skopeo: Check the minor version of skopeo

It has to be greater than 1.13.x.

Signed-off-by: Suraj Deshmukh <suraj.deshmukh@microsoft.com>

surajssd commented:
Merging this PR, any further changes we can fix in subsequent PRs.
